bin/configure-tower-project.py

```diff
@@ -0,0 +1,455 @@
+                "fullName": "Sage Bionetworks",
+                "description": None,
+                "location": None,
+                "website": None,
+                "logo": None,
+            },
+            "logoId": None,
+        }
+        response = self.send_tower_request("POST", endpoint, json=data)
+        return response["organization"]["orgId"]
+
+    def create_tower_workspace(self) -> Tuple[int, int]:
+        """Create a Tower workspace under the 'Sage Bionetworks' organization
+
+        Returns:
+            Tuple[int, int]: A pair of integer IDs correspond to the
+            'Sage Bionetworks' organization and the project workspace
+        """
+        # Get or create 'Sage Bionetworks' organization
+        org_id = self.create_tower_organization()
+        # Check if the project workspace already exists
+        endpoint = f"/orgs/{org_id}/workspaces"
+        response = self.send_tower_request("GET", endpoint)
+        for workspace in response["workspaces"]:
+            if workspace["name"] == self.stack_name:
+                return org_id, workspace["id"]
+        # Otherwise, create a new project workspace under the organization
+        data = {
+            "workspace": {
+                "name": self.stack_name,
+                "fullName": self.stack_name,
+                "description": None,
+                "visibility": "PRIVATE",
+            }
+        }
+        response = self.send_tower_request("POST", endpoint, json=data)
+        return org_id, response["workspace"]["id"]
+
+    def create_tower_credentials(self, project_stack: dict, ws_id: int) -> int:
+        """Create entry for Forge credentials under the given workspace
+
+        Args:
+            project_stack (dict): Outputs from the Tower project CFN stack
+            ws_id (int): Identifier for the project workspace
+
+        Returns:
+            int: Identifier for the Forge credentials entry
+        """
+        # Check if Forge credentials have already been created for this project
+        endpoint = "/credentials"
+        params = {"workspaceId": ws_id}
+        response = self.send_tower_request("GET", endpoint, params=params)
+        for cred in response["credentials"]:
+            if cred["name"] == self.stack_name:
+                assert cred["provider"] == "aws"
+                assert cred["deleted"] is None
+                return cred["id"]
+        # Otherwise, create a new credentials entry for the project
+        secret_arn = project_stack["TowerForgeServiceUserAccessKeySecretArn"]
+        credentials = self.get_secret_value(secret_arn)
+        data = {
+            "credentials": {
+                "name": self.stack_name,
+                "provider": "aws",
+                "keys": {
+                    "accessKey": credentials["aws_access_key_id"],
+                    "secretKey": credentials["aws_secret_access_key"],
+                    "assumeRoleArn": project_stack["TowerForgeServiceRoleArn"],
```

Comment:
does only nextlfow app have access to these creds or do humans have access as well?

Reply:
Previously, humans had access to this secret by virtue of the sandbox IAM roles listed in the Sceptre config. 

We have now restricted access to the admin roles on the AWS account. That's why we are leveraging the admin CI role here to pull the secret from Secrets Manager and store them in Tower.